src/shims/x86/sse42.rs

```diff
@@ -0,0 +1,454 @@
+use rustc_middle::mir;
+use rustc_middle::ty::layout::LayoutOf as _;
+use rustc_middle::ty::Ty;
+use rustc_span::Symbol;
+use rustc_target::abi::Size;
+use rustc_target::spec::abi::Abi;
+
+use crate::*;
+
+/// A constant for scrutinized the immediate byte provided
+/// to the string comparison intrinsics. It distinuishes between
+/// 16-bit integers and 8-bit integers.
```

Comment:
I can't parse the first sentence here... what are you trying to say?
From the use of the constant it seems like this is a bitmask? That would be worth saying.

Reply:
Yes, these constats are bitmasks. This is now documented.